src/main/java/com/bobocode/hoverla/bring/context/ConfigBasedBeanDefinition.java

```diff
@@ -0,0 +1,172 @@
+     * @throws BeanDefinitionConstructionException when config class of the instance passed is not marked as {@link Configuration}
+     *                                             or bean method is not marked as {@link Bean}
+     * @throws NullPointerException                when any of arguments passed is null
+     */
+    public ConfigBasedBeanDefinition(Object configInstance, Method beanMethod) {
+        validate(configInstance, beanMethod);
+
+        log.debug("Creating {} from method '{}'", ConfigBasedBeanDefinition.class.getSimpleName(), beanMethod);
+        this.configInstance = configInstance;
+        this.beanMethod = beanMethod;
+
+        log.trace("Getting name for bean");
+        this.name = getName(beanMethod);
+        log.trace("The name is '{}'", name);
+
+        log.trace("Getting type for '{}' bean", name);
+        this.type = getType(beanMethod);
+        log.trace("The type is '{}'", type);
+
+        log.trace("Getting dependencies for '{}' bean", name);
+        this.dependencies = getDependencies(beanMethod);
+        log.trace("Dependencies for '{}' bean are {}", name, dependencies);
+    }
+
+    private static void validate(Object configInstance, Method method) {
+        log.debug("Validating arguments passed to create ConfigBased");
+        Objects.requireNonNull(configInstance, "Configuration class instance is null");
+        Objects.requireNonNull(method, "Configuration method to create bean");
+
+        checkClassMarkedAsConfig(configInstance);
+        checkMethodMarkedAsBean(method);
+    }
+
+    private static void checkClassMarkedAsConfig(Object configInstance) {
+        if (!configInstance.getClass().isAnnotationPresent(Configuration.class)) {
+            throw new BeanDefinitionConstructionException(
+                    "Configuration class instance passed is not marked as @%s".formatted(Configuration.class.getSimpleName())
+            );
+        }
+    }
+
+    private static void checkMethodMarkedAsBean(Method method) {
+        if (!method.isAnnotationPresent(Bean.class)) {
+            throw new BeanDefinitionConstructionException(
+                    "Configuration method to create bean is not marked as @%s".formatted(Bean.class.getSimpleName())
+            );
+        }
+    }
+
+    private static String getName(Method beanMethod) {
+        Bean annotation = beanMethod.getAnnotation(Bean.class);
+        String beanName = annotation.name();
+        if (beanName.isEmpty()) {
+            return beanMethod.getName();
+        }
+        return beanName;
+    }
+
+    private static Class<?> getType(Method beanMethod) {
+        return beanMethod.getReturnType();
+    }
+
+    private static Map<String, Class<?>> getDependencies(Method beanMethod) {
+        return Arrays.stream(beanMethod.getParameters())
+                     .collect(toMap(
+                             ConfigBasedBeanDefinition::getParameterName,
```

Comment:
What is the point of splitting `toMap()` here? 
I would say `toMap(ConfigBasedBeanDefinition::getParameterName, Parameter::getType)` is well-readable

Reply:
I don't like it to be one long functional style line

Reply:
I'll go with your approach

Reply:
Aligned